### PR TITLE
Python3 port

### PR DESCRIPTION
**Pull request by icemac**
I created a PR of this branch to increase its visibility.

@stephan-hof Is this branch ready to merge?

### CODE REVIEW

File: buildout.cfg [hunk 1]
```diff
@@ -9,11 +9,14 @@ parts =
 
 [versions]
 Products.PythonScripts =
+AccessControl =
+RestrictedPython =
```
Comment (tseaver) [inline buildout.cfg]: Shouldn't we be relying on the now-released versions published via https://raw.githubusercontent.com/zopefoundation/Zope/master/versions.cfg (or even better, https://zopefoundation.github.io/Zope/releases/master)?

File: tox.ini [hunk 2]
```diff
@@ -1,16 +1,63 @@
 [tox]
 envlist =
-   py27,
-   py34,
-   py35,
-   py36,
+    # flake8,
```
Comment (tseaver) [inline tox.ini]: Why are we checking in commented-out environments?

File: tox.ini [hunk 3]
```diff
@@ -1,16 +1,63 @@
...
+
+skip_missing_interpreters = False
 
 [testenv]
+usedevelop = True
+extras =
+    develop
+    test
+
+install_command = pip install --no-cache-dir --no-binary=:all: {opts} {packages}
```
Comment (tseaver) [inline tox.ini]: Needs a comment explaining why we are overriding `install_command`.

File: tox.ini [hunk 4]
```diff
@@ -1,16 +1,63 @@
...
     {envbindir}/buildout -c {toxinidir}/buildout.cfg buildout:directory={envdir} buildout:develop={toxinidir} bootstrap
     {envbindir}/buildout -c {toxinidir}/buildout.cfg buildout:directory={envdir} buildout:develop={toxinidir} install test
-    {envbindir}/test
+    coverage run {envbindir}/test --all {posargs:-vc}
 skip_install = true
 deps =
     setuptools==33.1.1
     zc.buildout
+setenv =
+    PIP_NO_CACHE = 1
```
Comment (tseaver) [inline tox.ini]: Needs a comment explaining why.

File: tox.ini [hunk 5]
```diff
@@ -1,16 +1,63 @@
...
+[testenv:isort-apply]
+basepython = python2.7
+deps = isort
+commands =
+    isort --apply --recursive {toxinidir}/src {toxinidir}/tests {posargs}
+
+[testenv:flake8]
+basepython = python2.7
+deps = flake8
+commands = flake8 --doctests src tests setup.py
```
Comment (tseaver) [inline tox.ini]: These three testenv's should be either removed here or uncommented above.

**Review by tseaver — CHANGES_REQUESTED** [review]

**Comment by tseaver** [comment]
I would find it easier to review if the PR were rebased, with at least some commits squashed, to make  reviewing by commit easier.

**Comment by stephan-hof** [comment]
I'm sorry for all this small commits, I just committed on every test I fixed. I thought it is easier to review. I have no problem in squashing/rebasing them into a single commit.
Unfortunately rebasing is a disruptive operation, history is changed. So when it's done, other people having local (not pushed) commits get into trouble when pulling again.
So I'm reluctant to do it right away. It looks that other people like icemac are working on that branch as well and I don't want to destroy their local branches.
 
Another thing where I just need confirmation. The target python versions are
* python2 = 2.7
* python 3 >= 3.4
right ?

I'm not sure what should happen with the tox.ini file. Right now travis is using just buildout and there are no issues. I would vote for removing the file from this branch to work on it in a separate pull request. IMHO this flake8/coverage is an independent thing and not strongly required for the python3 port.

**Comment by icemac** [comment]
@tseaver I think I have addressed all your comments regarding tox.ini and buildout.cfg.

**Review by tseaver — APPROVED** [review]

**Pull request merged**